### PR TITLE
Feat/naming pipeline

### PR DESCRIPTION
**Pull request by jsome-nets**
Renaming a pipeline caused serious issues when running said pipeline. The name is used as part of resources belonging to the module, renaming caused many of them to be _replaced_ instead of updating in-place. This ended up deleting most of the resources mid-run.

This PR tries to make pipeline survive a renaming process by ignoring the name change on the resources that would require replacing.

### CODE REVIEW

**Review by julei-nets — COMMENTED** [review]
If name is ignored; then after initial resource creation, name changes ignored which results that you're unable to rename resources. I don't know how well this would work in reality.

How about lifecycle create before destroy?

**Comment by jsome-nets** [comment]
tldr; renaming pipeline is bad.

> If name is ignored; then after initial resource creation, name changes ignored which results that you're unable to rename resources. I don't know how well this would work in reality.

That is true, but in this manner the pipeline is in functioning state with only subset of resources renamed.

> How about lifecycle create before destroy?

Unfortunately that doesn't work. Let's say that we have pipeline `example` that gets renamed to `network`. Pipeline `example` creates a new pipeline `network` with create before destroy -config. The next step after creating is destroying the pipeline `example`. And right then and there the execution of the pipeline stops; regardless of the other changes that were supposed to happen. Same sort of effect is with removing/renaming the IAM roles/policies the pipeline uses.

Got to look into other options to prevent renaming. Creating a copy of the pipeline with new name could be the easiest solution. Have to look into could we be able to prevent users from renaming their pipelines.

**Comment by julei-nets** [comment]
Will close this as it's not suitable for v3 for the module. if rename support is ever desired, we'll design a fresh v3-specific change deliberately rather than carrying this forward.
